### PR TITLE
[C#]Building A Model using Code First Migration

### LINKED ISSUE #5917: Building a Model Using Code-first Workflow in .NET Core Web API.
### Topic Suggestion
Building a Model Using Code-first Workflow in .NET Core Web API.

#### Proposed title of the article
Your title should be descriptive of the article/tutorial. Be Specific. 

Use keyword research to improve your article's SEO.

#### Proposed article introduction
If you need to access data from your database, which you often will, you need a persistent framework. With the persistent framework, you will be able to load data or retrieve data from your database. Entity Framework is a persistent framework commonly used in .Net applications. 

#### Key takeaways 
What are the 3-5 most important things the reader should understand or be able to do after reading this article?
Use this area to get your ideas down on the bulk of your article or tutorial.

1. Adding new Classes.
          - The best way to name migrations.
          - How to use an already existing migration to do another similar migration.
          - How to populate a database table with data using code-first.
2. Modifying an existing class.
         - Ways one can override the conventions to modify database tables in EF code first migration
         - Different Methods applied during modification
3. Deleting an existing class.
         - How to use Entity framework code-first to preserve data before deleting a table.
4. Recover from mistakes.

#### Article quality
This article will go in-depth in terms of explaining the entity framework concepts. Most articles just explain the basics of creating, modifying, and deleting classes,  This article will go past that and explain in detail how migration works when these operations are applied and the changes one can do to the migration class to add, update and delete records in the database. By the end, the reader will be equipped all round on working with EF using .Net Core. Most articles use .NET MVC 5  to explain the concepts. 

#### References
N/A

### PR DESCRIPTION
**Pull request by munyite-megzs**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [X] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #5917

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by munyite-megzs** [comment]
Hello @srishilesh. I have gone through my article one more time. Hope to get a reviewer soon.

**Comment by srishilesh** [comment]
> Hello @srishilesh. I have gone through my article one more time. Hope to get a reviewer soon. 

@munyite-megzs 

That's awesome.

As soon as a reviewer completes existing PRs in backlog, they'll pick up a new PR. Please wait patiently until a reviewer pings you.

Thanks.

**Comment by Kayere** [comment]
Hello @munyite-megzs,

 Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
[[C#]Building A Model using Code First Migration #6483 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/8059030/C.Building.A.Model.using.Code.First.Migration.6483.Copyleaks.Report.pdf)

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.

Be sure to see our resources page to see more info on plagiarism and what is considered as such.

**Comment by munyite-megzs** [comment]
@Kayere  thank you. I have gone through the article again. However, I have one concern. There are some areas where there are similarities with other articles, though the context is totally different. Especially in some code methods. This is because that is the standard way it should be written. Is this still considered plagiarism?

File: content/articles/building-model-using-code-first/index.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-There are three workflows to build a domain model using Entity Framework. These are, `database first`, `Code first`, and `model first`.
+There are three workflows used to build a domain model using Entity Framework. These are, `database first`, `Code first`, and `model first`.
 
 With the `database first`, we start with the database. We design our tables using visual designers. The EntityFramework generates the domain classes based on the database.
```
Comment (Jethro-magaji) [inline content/articles/building-model-using-code-first/index.md]: These frameworks shouldn't all begin with the word "With". Find a way to a make the sentences continuous.

File: content/articles/building-model-using-code-first/index.md [hunk 2]
```diff
@@ -65,17 +67,17 @@ With this, we now have our API project. We now need to configure a few things in
 
 4. On the menu that appears, navigate to `New Folder` and click. Name the folder `Models`. 
 
-We will have four classes in our `Models` folder. These are, `Course`, `Author`, `Tag` and `CourseLevel`. The `CourseLevel` will be an enumeration. Let's have a closer look at each one. 
+We will have four classes in our `Models` folder. These are, `Course.cs`, `Author.cs`, `Tag.cs` and `CourseLevel.cs`. The `CourseLevel` will be an enumeration. Let's have a closer look at each one. 
 
-The course class has a few properties. These include the `Id`, `Title`, `Description`, `Level`, ` FullPrice`, `Author`(which is a navigation property), and a list of `Tags`. A course can have many tags. To create the class
+The `Course.cs` class has a few properties. These include the `Id`, `Title`, `Description`, `Level`, ` FullPrice`, `Author`(which is a navigation property), and a list of `Tags`. A course can have many tags. To create the class
 
 1. In the solution explorer, right-click on the `Models` folder.
```
Comment (Jethro-magaji) [inline content/articles/building-model-using-code-first/index.md:74]: I suggest add images here to guide your readers,

File: content/articles/building-model-using-code-first/index.md [hunk 3]
```diff
@@ -604,11 +607,11 @@ With this, let us run this migration on the database.
 
 ### Adding new classes.
 
-In visual studio, we are going to create a new class called `Category`. We will assign each course to a category. E.g. web development, mobile development, programming language, etc. 
+In visual studio, we are going to create a new class called `Category.cs`. We will assign each course to a category. E.g. web development, mobile development, programming language, etc.
```
Comment (Jethro-magaji) [inline content/articles/building-model-using-code-first/index.md:610]: You could shorten "visual studio" to "Vs code"

**Review by Jethro-magaji — CHANGES_REQUESTED** [review]
Hi! @munyite-megzs 
Well done for such an amazing article. However, there're some corrections you need to make.
Please, run your article on tools like Grammarly and ProWritingAid for better editing. This would save the review time to review your work.
Thanks.

File: content/articles/building-model-using-code-first/index.md (hunk 3 above)
Comment (munyite-megzs) [reply content/articles/building-model-using-code-first/index.md:612]: I am using visual studio and not visual studio code

**Comment by munyite-megzs** [comment]
Hello @Jethro-magaji ,
I have done the changes as requested, thank you.

**Review by Jethro-magaji — APPROVED** [review]
Hi! @munyite-megzs 
Well done for such an amazing article, please take note of these corrections so that you can minimize revisions in your next article.
Thanks.

**Comment by munyite-megzs** [comment]
@Jethro-magaji  this is well noted. Thank you

**Review by louisefindlay23 — CHANGES_REQUESTED** [review]
Your image files names and Application Names are not consistent. Also, there are grammatical issues and typos. I have fixed the image file names for you and some of the grammatical and typos but it is your responsibility to observe my edits and replicate them throughout your article.

**Comment by hectorkambow** [comment]
Will be closing this PR to allow the author the time to properly revise the articles where needed (on their own forked repo).

**Pull request closed without merging**